addon/internal-session.js

```diff
@@ -44,59 +43,80 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
   },
 
   restore() {
-    return new Ember.RSVP.Promise((resolve, reject) => {
-      let restoredContent   = this.store.restore();
-      let { authenticator } = (restoredContent.authenticated || {});
-      if (!!authenticator) {
-        delete restoredContent.authenticated.authenticator;
-        getOwner(this).lookup(authenticator).restore(restoredContent.authenticated).then((content) => {
+    return new RSVP.Promise((resolve, reject) => {
+      this._callStoreAsync('restore').then((restoredContent) => {
+        let { authenticator } = (restoredContent.authenticated || {});
+        if (!!authenticator) {
+          delete restoredContent.authenticated.authenticator;
+          getOwner(this).lookup(authenticator).restore(restoredContent.authenticated).then((content) => {
+            this.set('content', restoredContent);
+            this._setup(authenticator, content).then(resolve, reject);
+          }, () => {
+            Ember.Logger.debug(`The authenticator "${authenticator}" rejected to restore the session - invalidating…`);
+            this.set('content', restoredContent);
+            this._clear().then(resolve, reject);
+          });
+        } else {
+          delete (restoredContent || {}).authenticated;
           this.set('content', restoredContent);
-          this._setup(authenticator, content);
-          resolve();
-        }, () => {
-          Ember.Logger.debug(`The authenticator "${authenticator}" rejected to restore the session - invalidating…`);
-          this.set('content', restoredContent);
-          this._clear();
-          reject();
-        });
-      } else {
-        delete (restoredContent || {}).authenticated;
-        this.set('content', restoredContent);
-        this._clear();
-        reject();
-      }
+          this._clear().then(resolve, reject);
+        }
+      }, () => {
+        this._clear().then(resolve, reject);
+      });
     });
   },
 
+  _callStoreAsync(method, ...params) {
+    let result = this.store[method].apply(this.store, params);
+    if (typeof result === 'undefined' || typeof result.then === 'undefined') {
+      Ember.deprecate(`Ember Simple Auth: Synchronous stores have been deprecated. Make sure your custom store's ${method} method returns a promise.`, false, {
+        id: `ember-simple-auth.session-store.synchronous-${method}`,
+        until: '2.0.0'
+      });
```

Comment:
yes indeed, `Ember.deprecate` already exists as far as github goes back and just ignores the third argument in that case.